app/controllers/automated_tests_controller.rb

```diff
@@ -114,8 +114,8 @@ def populate_autotest_manager
 
     test_specs = autotest_settings_for(assignment)
     assignment_data = assignment.assignment_properties.attributes.slice(*required_params.map(&:to_s))
-    assignment_data['token_start_date'] ||= Time.current
-    assignment_data['token_start_date'] = assignment_data['token_start_date'].iso8601
+    assignment_data['token_period'] = assignment_data['token_period'].to_i
```

Comment:
This change isn't in the scope of this PR (I'm also not sure what problem it is fixing).